Build failure: diffoscope

### Nixpkgs version

- Unstable (26.11)

### Steps to reproduce

`nix-build -A diffoscope`

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/333219392

### Relevant log output

```console
=================================== FAILURES ===================================
________________________________ test_metadata _________________________________

differences = [<Difference zipinfo {} -- zipinfo {} []>, <Difference zipdetails --redact --walk --utc {} -- zipdetails --redact --walk --utc {} []>, <Difference dir/text -- dir/text []>]

    @skip_unless_tools_exist("zipinfo", "zipdetails")
    @skip_unless_tool_is_at_least("zipdetails", zipdetails_version, "4.004")
    @skip_unless_tool_is_at_least("perl", io_compress_zip_version, "2.212")
    def test_metadata(differences):
        assert_diff(differences[0], "zip_zipinfo_expected_diff")
>       assert_diff(differences[1], "zip_zipdetails_expected_diff")

tests/comparators/test_zip.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

difference = <Difference zipdetails --redact --walk --utc {} -- zipdetails --redact --walk --utc {} []>
filename = 'zip_zipdetails_expected_diff'

    def assert_diff(difference, filename):
        # Assign seen and expected values to local variables to improve contextual
        # information in failed tests.
        seen = difference.unified_diff
        expected = get_data(filename)
>       assert seen == expected
               ^^^^^^^^^^^^^^^^
E       AssertionError

tests/utils/data.py:64: AssertionError
______________________________ test_extra_fields _______________________________

differences2 = [<Difference bsdtar -tvf {} -- bsdtar -tvf {} []>, <Difference zipdetails --redact --walk --utc {} -- zipdetails --redact --walk --utc {} []>]

    @skip_unless_tools_exist("zipinfo", "bsdtar", "zipdetails")
    @skip_unless_tool_is_at_least("perl", io_compress_zip_version, "2.212")
    def test_extra_fields(differences2):
        assert_diff(differences2[0], "zip_bsdtar_expected_diff")
>       assert_diff(differences2[1], "zip2_zipdetails_expected_diff")

tests/comparators/test_zip.py:102: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

difference = <Difference zipdetails --redact --walk --utc {} -- zipdetails --redact --walk --utc {} []>
filename = 'zip2_zipdetails_expected_diff'

    def assert_diff(difference, filename):
        # Assign seen and expected values to local variables to improve contextual
        # information in failed tests.
        seen = difference.unified_diff
        expected = get_data(filename)
>       assert seen == expected
               ^^^^^^^^^^^^^^^^
E       AssertionError

tests/utils/data.py:64: AssertionError
=========================== short test summary info ============================
FAILED tests/comparators/test_zip.py::test_metadata - AssertionError
FAILED tests/comparators/test_zip.py::test_extra_fields - AssertionError
= 2 failed, 745 passed, 33 skipped, 6 deselected, 1 xfailed, 2 xpassed in 57.32s =
error: Cannot build '/nix/store/r73k5wscgdxfjjghk60xyc2y3xiyysvm-diffoscope-321.drv'.
       Reason: builder failed with exit code 1.
       Output paths:
         /nix/store/7437kv1i40ragcx3zri8nw1c7wlspd39-diffoscope-321-man
         /nix/store/baq6c9jha6kwysss2m0r9d4pll3ir0n2-diffoscope-321
         /nix/store/l6jk4sglv6fr0p1j2s07lpwgwi27csr1-diffoscope-321-dist
       Last 25 log lines:
       >     @skip_unless_tool_is_at_least("perl", io_compress_zip_version, "2.212")
       >     def test_extra_fields(differences2):
       >         assert_diff(differences2[0], "zip_bsdtar_expected_diff")
       > >       assert_diff(differences2[1], "zip2_zipdetails_expected_diff")
       >
       > tests/comparators/test_zip.py:102:
       > _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
       >
       > difference = <Difference zipdetails --redact --walk --utc {} -- zipdetails --redact --walk --utc {} []>
       > filename = 'zip2_zipdetails_expected_diff'
       >
       >     def assert_diff(difference, filename):
       >         # Assign seen and expected values to local variables to improve contextual
       >         # information in failed tests.
       >         seen = difference.unified_diff
       >         expected = get_data(filename)
       > >       assert seen == expected
       >                ^^^^^^^^^^^^^^^^
       > E       AssertionError
       >
       > tests/utils/data.py:64: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/comparators/test_zip.py::test_metadata - AssertionError
       > FAILED tests/comparators/test_zip.py::test_extra_fields - AssertionError
       > = 2 failed, 745 passed, 33 skipped, 6 deselected, 1 xfailed, 2 xpassed in 57.32s =
       For full logs, run:
         nix log /nix/store/r73k5wscgdxfjjghk60xyc2y3xiyysvm-diffoscope-321.drv
```

### Additional context

Same thing happens on 26.05. Last working commit is a0374025a863d007d98e3297f6aa46cc3141c2f0.

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.35, NixOS, 26.05 (Yarara), 26.05pre-git`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.34.7`
 - nixpkgs: `not found`

### Notify maintainers

@danielfullmer @mdaniels5757

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+label%3A%220.kind%3A+build+failure%22).
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.
- [x] I assert that I have read the [automation/AI policy](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#automationai-policy) and that this issue report complies with it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/di/diffoscope/fix-tests-with-zipdetails-4.006.patch
+++ b/pkgs/by-name/di/diffoscope/fix-tests-with-zipdetails-4.006.patch
@@ -1,0 +1,97 @@
+From fade8d04bfdbf473f3930feba7183957372e7fa7 Mon Sep 17 00:00:00 2001
+From: Michael Daniels <mdaniels5757@gmail.com>
+Date: Sun, 28 Jun 2026 16:20:43 -0400
+Subject: [PATCH] fix tests with zipdetails 4.006
+
+---
+ tests/comparators/test_zip.py            |  3 ++-
+ tests/data/zip2_zipdetails_expected_diff |  2 +-
+ tests/data/zip_zipdetails_expected_diff  | 10 +++++-----
+ 3 files changed, 8 insertions(+), 7 deletions(-)
+
+diff --git a/tests/comparators/test_zip.py b/tests/comparators/test_zip.py
+index 75ec38be..ecff1930 100644
+--- a/tests/comparators/test_zip.py
++++ b/tests/comparators/test_zip.py
+@@ -81,7 +81,7 @@ def differences2(zip1, zip3):
+ 
+ 
+ @skip_unless_tools_exist("zipinfo", "zipdetails")
+-@skip_unless_tool_is_at_least("zipdetails", zipdetails_version, "4.004")
++@skip_unless_tool_is_at_least("zipdetails", zipdetails_version, "4.006")
+ @skip_unless_tool_is_at_least("perl", io_compress_zip_version, "2.212")
+ def test_metadata(differences):
+     assert_diff(differences[0], "zip_zipinfo_expected_diff")
+@@ -96,6 +96,7 @@ def test_compressed_files(differences):
+ 
+ 
+ @skip_unless_tools_exist("zipinfo", "bsdtar", "zipdetails")
++@skip_unless_tool_is_at_least("zipdetails", zipdetails_version, "4.006")
+ @skip_unless_tool_is_at_least("perl", io_compress_zip_version, "2.212")
+ def test_extra_fields(differences2):
+     assert_diff(differences2[0], "zip_bsdtar_expected_diff")
+diff --git a/tests/data/zip2_zipdetails_expected_diff b/tests/data/zip2_zipdetails_expected_diff
+index 291dca88..281cf6c5 100644
+--- a/tests/data/zip2_zipdetails_expected_diff
++++ b/tests/data/zip2_zipdetails_expected_diff
+@@ -5,7 +5,7 @@
+  #
+  0064 Extra ID #1           5455 (21589) 'Extended Timestamp [UT]'
+  0066   Length              0009 (9)
+- 0068   Flags               03 (3) 'Modification Access'
++ 0068   Flags               03 (3) 'Modification & Access'
+ -0069   Modification Time   558AB455 (1435153493) 'Wed Jun 24 13:44:53 2015'
+ -006D   Access Time         558AB45F (1435153503) 'Wed Jun 24 13:45:03 2015'
+ +0069   Modification Time   41414141 (1094795585) 'Fri Sep 10 05:53:05 2004'
+diff --git a/tests/data/zip_zipdetails_expected_diff b/tests/data/zip_zipdetails_expected_diff
+index 978c2583..50df2696 100644
+--- a/tests/data/zip_zipdetails_expected_diff
++++ b/tests/data/zip_zipdetails_expected_diff
+@@ -23,7 +23,7 @@
+  #
+  0064 Extra ID #1           5455 (21589) 'Extended Timestamp [UT]'
+  0066   Length              0009 (9)
+- 0068   Flags               03 (3) 'Modification Access'
++ 0068   Flags               03 (3) 'Modification & Access'
+ -0069   Modification Time   558AB455 (1435153493) 'Wed Jun 24 13:44:53 2015'
+ +0069   Modification Time   558AB474 (1435153524) 'Wed Jun 24 13:45:24 2015'
+  006D   Access Time         558AB45F (1435153503) 'Wed Jun 24 13:45:03 2015'
+@@ -85,7 +85,7 @@
+  #
+ -01BF Extra ID #1           5455 (21589) 'Extended Timestamp [UT]'
+ -01C1   Length              0005 (5)
+--01C3   Flags               03 (3) 'Modification Access'
++-01C3   Flags               03 (3) 'Modification & Access'
+ -01C4   Modification Time   558AB455 (1435153493) 'Wed Jun 24 13:44:53 2015'
+ -01C8 Extra ID #2           7875 (30837) 'Unix Extra type 3 [ux]'
+ -01CA   Length              000B (11)
+@@ -96,7 +96,7 @@
+ -01D3   GID                 000003E8 (1000)
+ +024E Extra ID #1           5455 (21589) 'Extended Timestamp [UT]'
+ +0250   Length              0005 (5)
+-+0252   Flags               03 (3) 'Modification Access'
+++0252   Flags               03 (3) 'Modification & Access'
+ +0253   Modification Time   558AB455 (1435153493) 'Wed Jun 24 13:44:53 2015'
+ +0257 Extra ID #2           7875 (30837) 'Unix Extra type 3 [ux]'
+ +0259   Length              000B (11)
+@@ -163,7 +163,7 @@
+  #
+ -020D Extra ID #1           5455 (21589) 'Extended Timestamp [UT]'
+ -020F   Length              0005 (5)
+--0211   Flags               03 (3) 'Modification Access'
++-0211   Flags               03 (3) 'Modification & Access'
+ -0212   Modification Time   558AB455 (1435153493) 'Wed Jun 24 13:44:53 2015'
+ -0216 Extra ID #2           7875 (30837) 'Unix Extra type 3 [ux]'
+ -0218   Length              000B (11)
+@@ -174,7 +174,7 @@
+ -0221   GID                 000003E8 (1000)
+ +029C Extra ID #1           5455 (21589) 'Extended Timestamp [UT]'
+ +029E   Length              0005 (5)
+-+02A0   Flags               03 (3) 'Modification Access'
+++02A0   Flags               03 (3) 'Modification & Access'
+ +02A1   Modification Time   558AB474 (1435153524) 'Wed Jun 24 13:45:24 2015'
+ +02A5 Extra ID #2           7875 (30837) 'Unix Extra type 3 [ux]'
+ +02A7   Length              000B (11)
+-- 
+2.54.0
+

--- a/pkgs/by-name/di/diffoscope/package.nix
+++ b/pkgs/by-name/di/diffoscope/package.nix
@@ -126,6 +126,8 @@ python.pkgs.buildPythonApplication rec {
     ./ignore_links.patch
     # Remove flags output from an OCaml test's diff, as it's Debian-specific
     ./remove-flags-from-ocaml-diff.patch
+    # https://salsa.debian.org/reproducible-builds/diffoscope/-/merge_requests/166
+    ./fix-tests-with-zipdetails-4.006.patch
   ];
 
   postPatch = ''

--- a/pkgs/by-name/di/diffoscope/package.nix
+++ b/pkgs/by-name/di/diffoscope/package.nix
@@ -109,12 +109,12 @@ in
 # Note: when upgrading this package, please run the list-missing-tools.sh script as described below!
 python.pkgs.buildPythonApplication rec {
   pname = "diffoscope";
-  version = "321";
+  version = "322";
   pyproject = true;
 
   src = fetchurl {
     url = "https://diffoscope.org/archive/diffoscope-${version}.tar.bz2";
-    hash = "sha256-M/rsyUGlJDpU3o1RMfaN9fNMpOn9Xpz2ydflPUXVhD4=";
+    hash = "sha256-dina2JdbLL/jfo4eMuUo62KggST95w0b7oonY86zjgk=";
   };
 
   outputs = [

